src/server/game/Entities/Player/Player.cpp

```diff
@@ -27594,4 +27594,124 @@ bool Player::IsPetDismissed()
     return false;
 }
 
+/* Premium character interaction.
+ * These functions will allow modules to access the new generic
+ * premium tables designed to help modules that interact with
+ * players and characters without the need to create other tables.
+ */
+int Player::GetCharacterPremiumLevel(uint32 guid)
+{
+    PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_PREMIUM_CHARACTER_PREMIUM_LEVEL);
+    stmt->setUInt32(0, GUID_LOPART(guid));
+
+    PreparedQueryResult result = CharacterDatabase.Query(stmt);
+
+    if (!result)
+        return 0;
+
+    int premium_level = (*result)[0].GetInt8();
+    return premium_level;
+}
+
+
+bool Player::CreateCharacterPremiumLevel(uint32 guid, int premiumLevel)
```

Comment:
premiumLevel can be set to a smaller size like int8 maybe? Depending on how high you need it.
GUID like said above is 64 bit.